prime-router/src/main/kotlin/Metadata.kt

```diff
@@ -306,6 +337,57 @@ class Metadata {
         return loadLookupTable(name, table)
     }
 
+    /**
+     * Load the database lookup tables.
+     */
+    @Synchronized
+    internal fun checkForDatabaseLookupTableUpdates() {
+        // Check for tables at intervals
+        if (tablelastCheckedAt.plusSeconds(tablePollInternalSecs.toLong()).isAfter(Instant.now()))
+            return
+        loadDatabaseLookupTables()
+        tablelastCheckedAt = Instant.now()
+    }
+
+    /**
+     * Update the database lookup tables that have changed versions and load any new tables.
+     */
+    @Synchronized
+    internal fun loadDatabaseLookupTables() {
+        logger.trace("Checking for database lookup table updates...")
+        val databaseTables = lookupTableStore.values.mapNotNull { if (it is DatabaseLookupTable) it else null }
+
+        try {
+            val activeTables = tableDbAccess.fetchTableList()
+            // Let's be paranoid and check if the API is not returning what we need.
+            if (activeTables.any { it.isActive == false })
+                error("Database lookup table list returned an inactive table.")
+
+            // Process existing tables
+            databaseTables.forEach { dbTable ->
```

Comment:
This is hard to follow as it's written.

maybe  the loop can be flipped?:

```
activeTables.forEach { table ->
  val existingTable = databaseTables[table.name]
  // if it's not in the list it's a new table.
  // check it's version.
```
then handle the cases where an existing table was removed separately?

Reply:
Maybe this when statement can help